From 381f27746641238fe3e6df3e21ca363dfb4ecbb8 Mon Sep 17 00:00:00 2001
From: Felix Radensky <felix.r@variscite.com>
Date: Mon, 4 May 2020 11:36:14 +0300
Subject: [PATCH 2/4] imx8mm: Allocate UART2 to M4 domain only if M4 is enabled

i.MX8MM ATF code unconditionally allocates UART2 and UART4 to M4 domain.
On Variscite i.MX8MM boards UART4 is used by Linux running on A53 domain.
Allocate only UART2 to M4 domain and only if M4 is enabled.
---
 plat/imx/imx8mm/imx8mm_bl31_setup.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/plat/imx/imx8mm/imx8mm_bl31_setup.c b/plat/imx/imx8mm/imx8mm_bl31_setup.c
index aa1c336..85725c0 100644
--- a/plat/imx/imx8mm/imx8mm_bl31_setup.c
+++ b/plat/imx/imx8mm/imx8mm_bl31_setup.c
@@ -308,10 +308,11 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 
 	bl31_tzc380_setup();
 
-	/* Assign M4 to domain 1 */
-	mmio_write_32(IMX_RDC_BASE + 0x204, 0x1);
-	mmio_write_32(IMX_RDC_BASE + 0x518, 0xfc);
-	mmio_write_32(IMX_RDC_BASE + 0x5A4, 0xf3);
+	/* Assign UART2 to domain 1 */
+	if (imx_is_m4_enabled()) {
+		mmio_write_32(IMX_RDC_BASE + 0x204, 0x1);
+		mmio_write_32(IMX_RDC_BASE + 0x5A4, 0xf3);
+	}
 
 #if defined (CSU_RDC_TEST)
 	csu_test();
-- 
2.7.4

